### PR TITLE
Theme

### PR DESCRIPTION
**Pull request by davidoskky**
## Types of changes

- [x] I have read the **CONTRIBUTING** document.
- [x] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [x] This is **NOT** translation related. (See [here](https://github.com/aminecmi/ReaderforSelfoss/pull/170#issuecomment-355715654))

I fixed the text colors of the sources pages and simplified the logic here and there using the xml.

### CODE REVIEW

File: app/src/main/java/apps/amine/bou/readerforselfoss/SourcesActivity.kt [hunk 1]
```diff
@@ -29,19 +29,22 @@ class SourcesActivity : AppCompatActivity() {
...
     override fun onCreate(savedInstanceState: Bundle?) {
         appColors = AppColors(this@SourcesActivity)
-
-        super.onCreate(savedInstanceState)
-        binding = ActivitySourcesBinding.inflate(layoutInflater)
-        val view = binding.root
-
-        setContentView(view)
+        if (PreferenceManager.getDefaultSharedPreferences(this).getBoolean("dark_theme", false)) {
+            setTheme(R.style.NoBarDark)
```
Comment (aminecmi) [inline app/src/main/java/apps/amine/bou/readerforselfoss/SourcesActivity.kt:33]: Doesn't this break the theme color selector ?

**Comment by aminecmi** [comment]
Just tested this.

I see two issues.

## Settings activity does not take the primary and accent colors from the theme itself.

image


## Tab bar secondary color isn't refreshed after exciting the settings activity (primary colors change as expected)

(The settings)
image

(The colors without refreshing the app)
image

**Review by aminecmi — CHANGES_REQUESTED** [review]
Needs changing to handle secondary color issues, and theme activity issue.

**Comment by davidoskky** [comment]
I see the issue, I'll fix this after working on #376

**Comment by davidoskky** [comment]
The settings now uses the selected colors.
The colors selected are applied after leaving the settings page.
I'm unable to apply them while in the settings...

**Comment by aminecmi** [comment]
@davidoskky There is still one last issue.

image

The source title and time in dark theme are unreadable as they are in black.

**Comment by davidoskky** [comment]
Right, I didn't notice that before.
This is now solved as well.

**Pull request merged**